File: core/src/main/java/org/maibot/core/thinking/ThinkingFlowFactory.java
package org.maibot.core.thinking;

import org.maibot.core.cdi.annotation.AutoInject;
import org.maibot.core.cdi.annotation.ObjectFactory;
import org.maibot.core.cdi.annotation.Value;

@ObjectFactory
public class ThinkingFlowFactory {
    private final int observationWindowSize;

    private String flowId;

    @AutoInject
    private ThinkingFlowFactory(
            @Value("thinking.observation_window_size") int observationWindowSize
    ) {
        this.observationWindowSize = observationWindowSize;
    }

    public ThinkingFlowFactory setFlowId(String flowId) {
        this.flowId = flowId;
        return this;
    }


    public ThinkingFlow build() {
        return new ThinkingFlow(
                this.observationWindowSize,
                this.flowId
        );
    }
}

File: core/src/main/java/org/maibot/core/thinking/ThinkingFlowFactory.kt
package org.maibot.core.thinking

import org.maibot.sdk.ioc.AutoInject
import org.maibot.sdk.ioc.ObjectFactory
import org.maibot.sdk.ioc.Value

@ObjectFactory
class ThinkingFlowFactory @AutoInject private constructor(
    @param:Value($$"${thinking.observation_window_size}") private val observationWindowSize: Int
) {
    private var flowId: String? = null

    fun setFlowId(flowId: String): ThinkingFlowFactory {
        this.flowId = flowId
        return this
    }


    fun build(): ThinkingFlow {
        return ThinkingFlow(this.observationWindowSize, this.flowId)
    }
}
